backend/interface.go

```diff
@@ -0,0 +1,33 @@
+package backend
+
+import (
+	"errors"
+	"io"
+	"io/fs"
+	"os"
+)
+
+var (
+	ErrIncorrectOpenMode = errors.New("disk file or device not open for write")
+	ErrNotSuitable       = errors.New("backing file is not suitable")
+)
+
+type File interface {
+	fs.File
+	io.ReaderAt
+	io.Seeker
+	io.Closer
+}
+
+type WritableFile interface {
+	File
+	io.WriterAt
+}
```

Comment:
I like how you defined the interfaces clearly here. Thank you.

Reply:
Glad you liked it!